### DISCUSSION TITLE
Smart Meter Interface - mqtt without unit

### DISCUSSION THREAD

**Discussion by grericht** [arendst/Tasmota#18658] [Q&A]
Hello,
I built an SmartMeterInterface and flashed it with Tasmota. Everything works fine except that the unit of the values is not published with per mqtt.

![Screenshot (101)](https://github.com/arendst/Tasmota/assets/128326089/9146226b-a33b-43fa-ae35-a90dcc594a7d)
![Screenshot (104)](https://github.com/arendst/Tasmota/assets/128326089/d1732b72-6866-4eb8-8c8c-5f5f4a348830)

I have attached pictures of my Tasmota plug for comparison. The unit also arrives there in the Home Assistant.
![Screenshot (103)](https://github.com/arendst/Tasmota/assets/128326089/01678ce9-4a54-4fd6-824d-f543b5eacb4d)
d?
![Screenshot (102)](https://github.com/arendst/Tasmota/assets/128326089/d6062f73-1cfc-4f38-8b8b-bca2998dfa71)

Can I set somewhere that the unit should also be transmitte

**Comment by sfromis** [comment]
No, the Tasmota setup is that units are only included in the JSON when Tasmota has the option of configuring different units for the same value, like temperature or atmospheric pressure. What you see with the energy monitoring plug is not that units are "arriving" in HA, but that the Tasmota integration knows what units are used for certain JSON keys. Including units for arbitrary field names in the JSON would take a new convention for how this is done.

Also see this thread: https://github.com/arendst/Tasmota/discussions/18545
